pkg/query/storeset.go

```diff
@@ -453,16 +466,29 @@ func (s *StoreSet) updateStoreStatus(store *storeRef, err error) {
 		status = *prev
 	}
 
-	status.LastError = err
+	storeLabelsSetHash, err := storepb.LabelSetsToHash(store.labelSets)
+	if err != nil {
+		level.Error(s.logger).Log("msg", "create hash from label sets failed", "err", err)
+	}
+
+	status.LastError = storeUnhealthyReasonError
 	status.LastCheck = time.Now()
+	s.storeStatus.WithLabelValues(
+		storepb.LabelSetsToString(store.labelSets),
+		storeLabelsSetHash,
+	).Set(0)
 
-	if err == nil {
+	if storeUnhealthyReasonError == nil {
 
 		mint, maxt := store.TimeRange()
 		status.LabelSets = store.LabelSets()
 		status.StoreType = store.StoreType()
 		status.MinTime = mint
 		status.MaxTime = maxt
+		s.storeStatus.WithLabelValues(
```

Comment:
Hm this is hiding the truth as there can be many store APIs with the same labelset. So it's not only 0 or 1. This status can mean status for many storeAPIs essentially. It has to be number gauge with + and - 

... and this is exactly what `thanos_store_nodes_grpc_connections` does right?

Reply:
Yea that's what I tried to explain here: 

> Because this MR introduces a bool metric based on label set. The bool can be set too True, because the 1st is healthy, and then to False, because the second store is unhealthy. And to fix this it needs to be instrumented with the address as label.